src/status_im/subs/wallet/screens/input_amount.cljs

```diff
@@ -0,0 +1,360 @@
+ :<- [:wallet/wallet-send-enabled-from-chain-ids]
+ :<- [:wallet/bridge-from-chain-ids]
+ (fn [[tx-type send-chain-ids bridge-chain-ids]]
+   (if (= tx-type :tx/bridge)
+     bridge-chain-ids
+     send-chain-ids)))
+
+(rf/reg-sub :send-input-amount-screen/from-enabled-networks
+ :<- [:wallet/wallet-send-tx-type]
+ :<- [:wallet/wallet-send-enabled-networks]
+ :<- [:wallet/bridge-from-networks]
+ (fn [[tx-type send-enabled-networks bridge-enabled-networks]]
+   (if (= tx-type :tx/bridge)
+     bridge-enabled-networks
+     send-enabled-networks)))
+
+(rf/reg-sub :send-input-amount-screen/token-by-symbol
+ :<- [:send-input-amount-screen/enabled-from-chain-ids]
+ :<- [:wallet/wallet-send-token]
+ (fn [[enabled-from-chain-ids
+       {token-symbol :symbol}]]
+   (rf/sub [:wallet/token-by-symbol
+            (str token-symbol)
+            enabled-from-chain-ids])))
+
+(rf/reg-sub :send-input-amount-screen/total-balance
+ :<- [:send-input-amount-screen/token-by-symbol]
+ (fn [{:keys [total-balance]}]
+   total-balance))
+
+(rf/reg-sub :send-input-amount-screen/conversion-rate
+ :<- [:wallet/wallet-send-token]
+ :<- [:profile/currency]
+ (fn [[token
+       currency]]
+   (-> token
+       :market-values-per-currency
+       currency
+       :price)))
+
+(rf/reg-sub :send-input-amount-screen/usd-conversion-rate
+ :<- [:wallet/wallet-send-token]
+ (fn [token]
+   (utils/token-usd-price token)))
+
+(rf/reg-sub :send-input-amount-screen/token-decimals
+ :<- [:wallet/wallet-send-token]
+ (fn [token]
+   (-> token
+       utils/token-usd-price
+       utils/one-cent-value
+       utils/calc-max-crypto-decimals)))
+
+(rf/reg-sub :send-input-amount-screen/max-decimals
+ :<- [:send-input-amount-screen/state]
+ :<- [:send-input-amount-screen/token-decimals]
+ (fn [[{:keys [crypto-currency?]}
+       token-decimals]]
+   (if crypto-currency? token-decimals 2)))
+
+(rf/reg-sub :send-input-amount-screen/upper-limit
+ :<- [:send-input-amount-screen/state]
+ :<- [:send-input-amount-screen/conversion-rate]
+ :<- [:send-input-amount-screen/usd-conversion-rate]
+ :<- [:send-input-amount-screen/total-balance]
+ (fn [[{:keys [crypto-currency?]}
+       conversion-rate
+       usd-conversion-rate
+       total-balance]]
+   (if crypto-currency?
+     (utils/cut-crypto-decimals-to-fit-usd-cents
+      total-balance
+      usd-conversion-rate)
+     (utils/cut-fiat-balance-to-two-decimals
+      (money/crypto->fiat total-balance conversion-rate)))))
+
+(rf/reg-sub :send-input-amount-screen/upper-limit-prettified
+ :<- [:send-input-amount-screen/state]
+ :<- [:profile/currency-symbol]
+ :<- [:wallet/wallet-send-token]
+ :<- [:send-input-amount-screen/conversion-rate]
+ :<- [:send-input-amount-screen/upper-limit]
+ (fn [[{:keys [crypto-currency?]}
+       currency-symbol
+       {token-symbol :symbol}
+       conversion-rate
+       upper-limit]]
+   (if crypto-currency?
+     (utils/prettify-crypto-balance
+      (or (clj->js token-symbol) "")
```

Comment:
These coercions using `clj->js`, or `money/bignumber` inside the subs seem like a smell. Many functions in the PR have some sort of coercion, which makes the code fragile. Wouldn't it be possible to decode the values exactly once, and write to the state?

Maybe this is just the result of too much data mangling for the UI at the ~wrong~ sub layer?

Reply:
Actually, if we pay attention to this piece of code (inherited). It's a bit weird because we transform to JS and wrap it on an `or` :thinking: , also what is `token-symbol` expected to be that we want it on JS?

I think this is something that we wanted to be `(name token-symbol)` if it is a keyword.

@vkjr What do you think `token-symbol` is?

Reply:
@ulisesmac, I remember trying to figure purpose of this conversion and as far as I understood `token-symbol` is a keyword. And `cljs->js` used here instead of `name` because `name` throws an error when argument is `nil`. So my guess is that one day developer ran into a problem with thrown exception because `token-symbol` appeared to be `nil` and created this workaround.

Reply:
My own lessons learned from the past are to decode/coerce data as early as possible if performance allows (it should usually because some code will decode eventually/soon). This way, the correct instances flow in the system, not primitive strings representing numbers. In terms of performance these type transformations in subs cause unnecessary work over and over, and there's a good chance other parts of the code base are doing the same transformations on the primitive types.

Just one idea: with malli we can describe transformers to decode any data https://github.com/metosin/malli?tab=readme-ov-file#value-transformation We used that in another CLJS project quite successfully.

The idea would be more or less like:

1. A response from an RPC request or signal arrives. The response contains primitive values representing dates and big numbers.
2. We need to store in the app-db as clj data.
3. We define a schema and call decode from the response.
4. We persist the result in the app-db, making sure all data types implemented the necessary interfaces, like what we did with `BigNumber` in `money.cljs`. This will allow re-frame to do equality checks correctly without any extra work from devs.

The only thing worth measuring is the performance aspect. I don't remember now if malli decoding is roughly as fast as hand-rolled code.

Reply:
I like the idea @ilmotta 

This will also work as documentation we can always check.
We need to check the performance aspect, but sounds great.

Reply:
BTW, this `clj->js` transformation is not related to a late data tranformation